client/me/purchases/remove-plan-dialog/get-plan-features.tsx

```diff
@@ -0,0 +1,257 @@
+import {
+	isMonthly,
+	isWpComBusinessPlan,
+	isWpComEcommercePlan,
+	isWpComPersonalPlan,
+	isWpComPremiumPlan,
+} from '@automattic/calypso-products';
+import { useTranslate } from 'i18n-calypso';
+
+export default function getPlanFeatures(
+	productSlug: string | undefined,
+	translate: ReturnType< typeof useTranslate >,
+	withinFirstYear: boolean,
+	hasDomain: boolean,
+	domainSlug: string
+): string[] {
+	if ( ! productSlug ) {
+		return [];
+	}
+
+	const isMonthlyPlan = isMonthly( productSlug );
+
+	/**
+	 * Define feature labels.
+	 */
+	const freeDomainFirstYear = String( translate( 'Your free domain for the first year' ) );
+	const freeDomain = String(
+		translate( 'Your free domain www.%(slug)s', {
+			args: { slug: domainSlug },
+		} )
+	);
+	const bestInClassHosting = String( translate( 'Best-in-class hosting' ) );
+	const addFreeSite = String( translate( 'An add-free site' ) );
+	const collectPayments = String( translate( 'The ability to collect payments' ) );
+	const emailCustomerSupport = String( translate( 'Unlimited customer support via email' ) );
+	const liveChat = String( translate( 'Access to live chat support' ) );
+	const earnAndRevenue = String( translate( 'The ability to earn and revenue' ) );
+	const premiumThemes = String( translate( 'Access to premium themes' ) );
+	const analyticsIntegration = String( translate( 'Google Analytics integration' ) );
+	const accessPlugins = String( translate( 'Access to more than 50,000 plugins' ) );
+	const seoTools = String( translate( 'Advanced SEO tools' ) );
+	const automatedBackups = String( translate( 'Automated site backupsand one-click restore' ) );
+	const sftpAndDatabase = String( translate( 'SFTP and Database access' ) );
+	const acceptPayments = String( translate( 'Accept payments in 60+ countries' ) );
+	const shippingCarriers = String( translate( 'Integration with top shipping carriers' ) );
+	const premiumDesign = String(
+		translate( 'Premium design options customized for online stores' )
+	);
+	const more = String( translate( 'and more' ) );
```

Comment:
I think the reason is that `translate()` can be used to create interpolated React components, so its return type is `TranslateResult` (this is similar to `ReactNode`). This function returns `string[]`, which is a sensible return value, since all of the `translate()` calls above will in fact only return strings, but TypeScript doesn't know that. The cast forces the results to be strings in case they are not. We do this in a lot of places ([example](https://github.com/Automattic/wp-calypso/blob/b472d87a530486b28f61a9944cb8b540a14478fc/client/my-sites/checkout/modal/index.tsx#L73)).

I learned recently that there's another way to do this which might be more semantic. The `translate()` method accepts an options object as its second parameter and that object can have the `textOnly` property. If set, TypeScript will know that the function will return a string.

https://github.com/Automattic/wp-calypso/blob/b472d87a530486b28f61a9944cb8b540a14478fc/client/components/jetpack/profile-dropdown/index.tsx#L65

So this line could be: 

```js
	const more = translate( 'and more', { textOnly: true } );
```

Reply:
But can't we return TranslateResult[] and avoid all the confusion? @sirbrillig

Reply:
Certainly, if you like. Sometimes that's the more useful fix.

Reply:
I leave that judgment call to you @ciprianimike :) I am ok either way too 👍

Reply:
@jdc91  I see this approach used all over the codebase so I would keep it as it is for consistency, this is the reason why I used it